docs/audit-log-filter-advanced.md

```diff
@@ -0,0 +1,847 @@
+                  "name": "general_sql_command.str",
+                  "value": "alter_user"
+                }
+              },
+              {
+                "field": {
+                  "name": "general_user.str",
+                  "value": "root"
+                }
+              }
+            ]
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+### OR Operations
+
+Log if any condition is true:
+
+```json
+{
+  "filter": {
+    "class": {
+      "name": "table_access",
+      "event": [
+        {
+          "name": "insert",
+          "log": {
+            "or": [
+              {
+                "field": {
+                  "name": "table_name.str",
+                  "value": "users"
+                }
+              },
+              {
+                "field": {
+                  "name": "table_name.str",
+                  "value": "payments"
+                }
+              }
+            ]
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+### NOT Operations
+
+Invert condition results:
+
+```json
+{
+  "filter": {
+    "class": {
+      "name": "general",
+      "event": [
+        {
+          "name": "status",
+          "log": {
+            "not": {
+              "field": {
+                "name": "general_sql_command.str",
+                "value": "select"
+              }
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+### Complex Logical Combinations
+
+Combine multiple logical operators:
+
+```json
+{
+  "filter": {
+    "class": {
+      "name": "table_access",
+      "event": [
+        {
+          "name": "update",
+          "log": {
+            "and": [
+              {
+                "or": [
+                  {
+                    "field": {
+                      "name": "table_name.str",
+                      "value": "users"
+                    }
+                  },
+                  {
+                    "field": {
+                      "name": "table_name.str",
+                      "value": "accounts"
+                    }
+                  }
+                ]
+              },
+              {
+                "not": {
+                  "field": {
+                    "name": "general_user.str",
+                    "value": "readonly_user"
+                  }
+                }
+              }
+            ]
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+## Predefined Variables and Functions
+
+### Using Predefined Variables
+
+Reference system variables in filter conditions:
+
+```json
+{
+  "filter": {
+    "class": {
+      "name": "connection",
+      "log": {
+        "field": {
+          "name": "connection_id",
+          "value": "1"
+        }
+      }
+    }
+  }
+}
+```
+
+### Using Predefined Functions
+
+Apply functions to field values:
+
+```json
+{
+  "filter": {
+    "class": {
+      "name": "general",
+      "event": [
+        {
+          "name": "status",
+          "log": {
+            "function": {
```

Comment:
Functions are used in a bit different context
For instance, here in one of the MTR test cases
https://github.com/percona/percona-server/blob/8.0/plugin/audit_log_filter/tests/mtr/t/filter_definition_replace_field.test#L20

Reply:
I suggest to use `string_find` function here. We can log only those `general|status` events in which `general_sql_command.str` starts with `show_` (all `SHOW XXX` statements)